src/nyx/features/image_moments.cpp

```diff
@@ -224,13 +207,33 @@ double ImageMomentsFeature::moment (const pixcloud& cloud, int p, int q)
     return sum;
 }
 
+/// @brief Calculates a spatial 2D moment of order q,p of ROI pixel cloud 'c' using real-valued intensities 'real_intens'
+double ImageMomentsFeature::moment (const pixcloud& c, const std::vector<float>& real_intens, int p, int q)
+{
+    double q_ = q, p_ = p, sum = 0;
+    size_t n = c.size();
+    for (size_t i=0; i<n; i++)
+    {
+        auto& pxl = c[i];
+        sum += real_intens[i] * pow(double(pxl.x - baseX), p_) * pow(double(pxl.y - baseY), q_);
+    }
+    return sum;
+}
+
 void ImageMomentsFeature::calcOrigins (const std::vector<Pixel2>& cloud)
 {
     double m00 = moment (cloud, 0, 0);
     originOfX = moment (cloud, 1, 0) / m00;
     originOfY = moment (cloud, 0, 1) / m00;
 }
 
+void ImageMomentsFeature::calcOrigins (const std::vector<Pixel2>& cloud, const std::vector<float>& real_valued_intensities)
+{

```

Comment:
Not seeing where `real_valued_intensities` is used.

Reply:
Fixed in the update